lib/split/persistence/dual_adapter.rb

```diff
@@ -22,22 +26,28 @@ def initialize(context)
           raise "Please configure :logged_out_adapter"
         end
 
-        if logged_in.call(context)
-          @adapter = logged_in_adapter.new(context)
-        else
-          @adapter = logged_out_adapter.new(context)
-        end
+        @logged_in = logged_in.call(context)
+        @logged_in_adapter = logged_in_adapter.new(context)
+        @logged_out_adapter = logged_out_adapter.new(context)
       end
 
-      def self.with_config(options={})
-        self.config.merge!(options)
-        self
+      def keys
+        @logged_in_adapter.keys + @logged_out_adapter.keys
```

Comment:
Return keys from both adapters can return duplicated keys. Other places internally iterate over keys so, this can lead to extra work.

I believe that it's better to keep the old behavior here, which is to return the keys from the current adapter (logged_in/logged_out).

Reply:
good catch, are you sure it's better `logged_in/logged_out` instead of `(logged_in + logged_out).uniq` ? I'm asking because I'm not sure if it can have some negative impacts

Reply:
I don't think it's going to have negative impacts(at least internally) if we can gate this behavior under a configuration flag I mentioned on the other comment, we can use the `uniq` version.